tests/utils.py

```diff
@@ -90,25 +92,45 @@ def __init__(self, private_key):
     def sign(self, message_hash):
         return sign(msg_hash=message_hash, priv_key=self.private_key)
 
-    async def send_transaction(self, account, to, selector_name, calldata, nonce=None):
+    async def send_transaction(self, account, to, selector_name, calldata, nonce=None, max_fee=0):
+        return await self.send_transactions(account, [(to, selector_name, calldata)], nonce, max_fee)
+
+    async def send_transactions(self, account, calls, nonce=None, max_fee=0):
         if nonce is None:
             execution_info = await account.get_nonce().call()
             nonce, = execution_info.result
 
-        selector = get_selector_from_name(selector_name)
-        message_hash = hash_message(
-            account.contract_address, to, selector, calldata, nonce)
+        mcalls = []
+        calls_with_selector = []
+        calldata = []
+        for i in range(len(calls)):
+            if len(calls[i]) != 3:
+                raise Exception("Invalid call parameters")
+            call = calls[i]
+            mcall = (call[0], get_selector_from_name(call[1]), len(calldata), len(call[2]))
+            mcalls.append(mcall)
+            calldata.extend(call[2])
+            calls_with_selector.append((call[0], get_selector_from_name(call[1]), call[2]))
```

Comment:
it would be nice to abstract this away for readability

Reply:
Agree. I've separated the 2 functionalities:
-  adding the selector in `calls` using `map` 
- going from `calls` to `mcalls` + `calldata` using a new utility method `from_call_to_mcall`